code/_onclick/item_attack.dm

```diff
@@ -149,7 +149,7 @@
 // Proximity_flag is 1 if this afterattack was called on something adjacent, in your square, or on your person.
 // Click parameters is the params string from byond Click() code, see that documentation.
 /obj/item/proc/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	SEND_SIGNAL(src, COMSIG_ITEM_AFTERATTACK, target, user, proximity_flag, click_parameters)
+	SEND_SIGNAL(target, COMSIG_ITEM_AFTERATTACK, src, user, target, proximity_flag, click_parameters)
```

Comment:
Switching the params here breaks the duct tape component, specifically sticking items to walls that have tape on them. It's currently the only other thing relying on `COMSIG_ITEM_AFTERATTACK`. You'll either need to fix duct tape as is, or change these params back, and find another way to fix the label component.

Reply:
> Switching the params here breaks the duct tape component, specifically sticking items to walls that have tape on them. It's currently the only other thing relying on `COMSIG_ITEM_AFTERATTACK`. You'll either need to fix the duct tape issue as is, or change these params back, and find another way to fix the label component.

I was not aware of that duct tape functionality. I shall look into it.